File: 卡码/代码随想录/JS/二叉树/04-二叉搜索树属性/LeetCode_0098.js
//98. 验证二叉搜索树
import TreeNode from "../TreeNode.js"

const isValidBST = function(root) {
  let pre=null;
  const traversal=node=>{
    if(!node) return true;
    const left=traversal(node.left);
    if(pre&&pre.val>=node.val) return false;
    pre=node;
    const right=traversal(node.right);
    return left&&right;
  }
  return traversal(root);
};

File: 卡码/代码随想录/Python/二叉树/04-二叉搜索树属性/LeetCode_0098.py
#98. 验证二叉搜索树
from typing import *
from ..TreeNode import *

class Solution:
  def __init__(self):
    self.pre:TreeNode=None
  def isValidBST(self, root: Optional[TreeNode]) -> bool:
    if not root:
      return True
    left:bool=self.isValidBST(root.left)
    if self.pre and self.pre.val>=root.val:
      return False
    self.pre=root
    right:bool=self.isValidBST(root.right)
    return left and right
